org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -2099,25 +2097,33 @@ private String getAnnotationType(IJavaProject javaProject, List<String> annotati
-		options.put(JavaCore.COMPILER_PB_NULL_ANNOTATION_INFERENCE_CONFLICT, defaultOptions.get(JavaCore.COMPILER_PB_NULL_ANNOTATION_INFERENCE_CONFLICT));
-	}
-
-	private void enableAnnotationNullAnalysis(Map<String, String> options, String nonnullType, String nullableType) {
-		options.put(JavaCore.COMPILER_ANNOTATION_NULL_ANALYSIS, "enabled");
-		options.put(JavaCore.COMPILER_NONNULL_ANNOTATION_NAME, nonnullType != null ? nonnullType : "");
-		options.put(JavaCore.COMPILER_NULLABLE_ANNOTATION_NAME, nullableType != null ? nullableType : "");
-		options.put(JavaCore.COMPILER_PB_NULL_REFERENCE, "warning");
-		options.put(JavaCore.COMPILER_PB_POTENTIAL_NULL_REFERENCE, "warning");
-		options.put(JavaCore.COMPILER_PB_NULL_SPECIFICATION_VIOLATION, "warning");
-		options.put(JavaCore.COMPILER_PB_NULL_ANNOTATION_INFERENCE_CONFLICT, "warning");
+	private Map<String, String> generateProjectNullAnalysisOptions(String nonnullType, String nullableType) {
+		Map<String, String> options = new HashMap<>();
+		if (nonnullType == null && nullableType == null) {
+			options.put(JavaCore.COMPILER_ANNOTATION_NULL_ANALYSIS, "disabled");
+			// set default values
+			Hashtable<String, String> defaultOptions = JavaCore.getDefaultOptions();
+			options.put(JavaCore.COMPILER_NONNULL_ANNOTATION_NAME, defaultOptions.get(JavaCore.COMPILER_NONNULL_ANNOTATION_NAME));
+			options.put(JavaCore.COMPILER_NULLABLE_ANNOTATION_NAME, defaultOptions.get(JavaCore.COMPILER_NULLABLE_ANNOTATION_NAME));
+			options.put(JavaCore.COMPILER_PB_NULL_REFERENCE, defaultOptions.get(JavaCore.COMPILER_PB_NULL_REFERENCE));
+			options.put(JavaCore.COMPILER_PB_POTENTIAL_NULL_REFERENCE, defaultOptions.get(JavaCore.COMPILER_PB_POTENTIAL_NULL_REFERENCE));
+			options.put(JavaCore.COMPILER_PB_NULL_SPECIFICATION_VIOLATION, defaultOptions.get(JavaCore.COMPILER_PB_NULL_SPECIFICATION_VIOLATION));
+			options.put(JavaCore.COMPILER_PB_NULL_ANNOTATION_INFERENCE_CONFLICT, defaultOptions.get(JavaCore.COMPILER_PB_NULL_ANNOTATION_INFERENCE_CONFLICT));
+		} else {
+			options.put(JavaCore.COMPILER_ANNOTATION_NULL_ANALYSIS, "enabled");
+			options.put(JavaCore.COMPILER_NONNULL_ANNOTATION_NAME, nonnullType != null ? nonnullType : "");
+			options.put(JavaCore.COMPILER_NULLABLE_ANNOTATION_NAME, nullableType != null ? nullableType : "");
+			options.put(JavaCore.COMPILER_PB_NULL_REFERENCE, "warning");
+			options.put(JavaCore.COMPILER_PB_POTENTIAL_NULL_REFERENCE, "warning");
+			options.put(JavaCore.COMPILER_PB_NULL_SPECIFICATION_VIOLATION, "warning");
+			options.put(JavaCore.COMPILER_PB_NULL_ANNOTATION_INFERENCE_CONFLICT, "warning");
+		}
+		return options;
+	}
+
+	private boolean isProjectNullAnalysisOptionsChanged(Map<String, String> current, Map<String, String> replacement) {
+		return !(Objects.equals(current.get(JavaCore.COMPILER_ANNOTATION_NULL_ANALYSIS), replacement.get(JavaCore.COMPILER_ANNOTATION_NULL_ANALYSIS))
```

Comment:
If all the values in the map are `String`, you can simply use `Map.equals()`.

Reply:
return first.entrySet().stream()
      .allMatch(e -> e.getValue().equals(second.get(e.getKey())));